python/feathub/processors/flink/table_builder/tests/test_flink_table_builder_derived_feature.py

```diff
@@ -512,3 +514,38 @@ def test_join_transform_with_zoned_timestamp(self):
         self.flink_table_builder.t_env.get_config().set_local_timezone(
             prev_local_timezone
         )
+
+    def test_bounded_left_table_join_unbounded_right_table(self):
+        source = DataGenSource(
+            name="source_1",
+            schema=Schema(["id", "val1", "time"], [Int64, Int64, String]),
+            timestamp_field="time",
+            timestamp_format="%Y-%m-%d %H:%M:%S",
+            keys=["id"],
+        )
+
+        source_2 = DataGenSource(
+            name="source_2",
+            schema=Schema(["id", "val2", "time"], [Int64, Int64, String]),
+            timestamp_field="time",
+            timestamp_format="%Y-%m-%d %H:%M:%S",
+            keys=["id"],
+        )
+
+        feature_view_1 = DerivedFeatureView(
+            name="feature_view_1",
+            source=source,
+            features=["source_2.val2"],
+            keep_source_fields=True,
+        )
+
+        built_feature_view = self.registry.build_features([source_2, feature_view_1])[1]
+
+        with self.assertRaises(FeathubException) as cm:
+            self.flink_table_builder.build(built_feature_view.get_bounded_view())
```

Comment:
Would it be simpler to define `source` as a bounded table instead of calling `get_bounded_view()` here?